web/ai_service.py

```
@@ -0,0 +1,141 @@
+        # Initialize Gemini if API key is available
+        if settings.USE_GEMINI:
+            self.gemini_api_key = settings.GEMINI_API_KEY
+            genai.configure(api_key=self.gemini_api_key)
+
+    def _format_code_block(self, code: str, language: str = "python") -> str:
+        """Format code blocks with syntax highlighting."""
+        try:
+            lexer = get_lexer_by_name(language)
+            formatter = HtmlFormatter(style='monokai', linenos=True)
+            highlighted = highlight(code, lexer, formatter)
+            return f'<div class="code-block">{highlighted}</div>'
```

Ruff (0.8.2)

Consider moving this statement to an `else` block

(TRY300)